app/(home)/activities/page.tsx

```diff
@@ -1,10 +1,147 @@
+"use client";
+import { useState } from "react";
 import PageHero from "@/components/shared/PageHero";
-import React from "react";
+import { upcomingEvents, events } from "@/libs/constant/events.constant";
+import EventCarousel from "@/components/EventCarousel";
+
+// Define types for event items
+interface EventItem {
+  title: string;
+  image: string;
+  alt: string;
+  des: string;
+}
+
+// Define props for the Modal component
+interface ModalProps {
+  image?: string;
+  alt?: string;
+  onClose: () => void;
+}
+
+// Modal component
+const Modal: React.FC<ModalProps> = ({ image, alt, onClose }) => {
```

Comment:
Separate the Modal component as  a file inside components, if planning to use the same modal component in multiple pages, then keep the component in shared